File: primes2.py
#!/usr/bin/env python
    
def make_isprime():
    memo = []
    def isprime(n):
        for prime in memo:
            if not n % prime: return False
        memo.append(n)
        return True
    return isprime

if __name__ == '__main__':
    import sys
    try:
        limit = int(sys.argv[1])
    except:
        print("{} LIMIT".format(sys.argv[0]))
    else:
        primes = filter(make_isprime(), range(2, limit+1))
        print(' '.join(map(str, primes)))

File: primes2.swift
func make_isprime() -> (Int -> Bool) {
    var memo = Int[]()
    func isprime(n: Int) -> Bool {
        for prime in memo {
            if n % prime == 0 { return false }
	}
        memo.append(n)
        return true
    }
    return isprime
}

func intifyarg1() -> Int? {
    if C_ARGC == 2 {
        let limitstr = String.fromCString(C_ARGV[1])
	if let limit = limitstr.toInt() {
	    return limit
        }
    }
    return nil
}

if let limit = intifyarg1() {
    for prime in filter(2...limit, make_isprime()) {
        print("\(prime) ")
    }
    println("")
} else {
    println("\(C_ARGV[0]) LIMIT")
}
